pkiinit/option/pkiOption.go

```diff
@@ -0,0 +1,125 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+package option
+
+import (
+	"errors"
+	"fmt"
+)
+
+// ExitCode is the code used for exit status
+type exitCode int
+
+const (
+	// normal exit code
+	normal exitCode = 0
+	// noOptionSelected exit code
+	noOptionSelected exitCode = 1
+	// exitWithError is exit code for error
+	exitWithError exitCode = -1
+)
+
+// PkiInitOption contains command line options for pki-init
+type PkiInitOption struct {
+	generateOpt bool
+	importOpt   bool
+	cacheOpt    bool
+	cacheCAOpt  bool
+}
+
+// NewPkiInitOption constructor to get options built for pki-init
+func NewPkiInitOption(options ...func(*PkiInitOption)) *PkiInitOption {
+	pkiInitOpt := &PkiInitOption{}
+	for _, opt := range options {
+		opt(pkiInitOpt)
+	}
+	return pkiInitOpt
+}
+
+// SetGenerate is the setter functional option for pki-init generate
+func SetGenerate(input bool) func(*PkiInitOption) {
+	return func(pkiInitOpton *PkiInitOption) {
+		pkiInitOpton.generateOpt = input
+	}
+}
+
+// SetImport is the setter functional option for pki-init import
+func SetImport(input bool) func(*PkiInitOption) {
+	return func(pkiInitOpton *PkiInitOption) {
+		pkiInitOpton.importOpt = input
+	}
+}
+
+// SetCache is the setter functional option for pki-init cache
+func SetCache(input bool) func(*PkiInitOption) {
+	return func(pkiInitOpton *PkiInitOption) {
+		pkiInitOpton.cacheOpt = input
+	}
+}
+
+// SetCacheCA is the setter functional option for pki-init cacheCA
+func SetCacheCA(input bool) func(*PkiInitOption) {
+	return func(pkiInitOpton *PkiInitOption) {
+		pkiInitOpton.cacheCAOpt = input
+	}
+}
+
+// ProcessOptions processes all the input options from caller
+func (pkiInitOpt *PkiInitOption) ProcessOptions() (int, error) {
+	if pkiInitOpt.NoOptionSelected() {
+		return int(noOptionSelected), errors.New("need at least one option, show help message")
+	}
+
+	statusCode, err := pkiInitOpt.executeOptions(
```

Comment:
US4404 only asked that the "help" option be supported.  The main event loop should be moved to another class, take pkiOption struct as an argument, and only implement "help" processing.

Reply:
yes, only help is supported or implemented. others are just template code and does not have the real functionality yet.